### PR TITLE
Added 'ident' command to identify to IRC services.

### PR DESCRIPTION
**Pull request by mikotohe2**
Added in convenient services identification. NickServ nick (in case it is not NickServ) and services password are stored in gchat-irc.config.yml.

### CODE REVIEW

**Comment by acffordyce973** [comment]
Getting invalid syntax when testing on Badnik.

**Comment by mikotohe2** [comment]
Huh. Maybe 'msg %s id' should be msg %s identify? I wonder if all networks support id in place of identify.

**Comment by acffordyce973** [comment]
Can confirm that alama's "ircBot.msg(ircServiceName, "identify %s" % (ircServicePass))" works as intended.

**Comment by mikotohe2** [comment]
That seems to work.

**Comment by alama** [comment]
Also, you may want to hook privmsg() function line 105 so if we get a msg from ircServiceName, fire the ident message

**Comment by mikotohe2** [comment]
That's what I was thinking of adding next, so the bot will auth automatically when using a registered nick.

File: proxy/plugins/GlobalChat.py [hunk 1]
```diff
@@ -390,4 +403,4 @@ def call_from_console(self):
                     client.get_handle().send_crypto_packet(TCPacket)
                 else:
                     client.get_handle().send_crypto_packet(SMPacket)
-        return "[GlobalChat] <Console> %s" % self.args[2:]
+        return "[GlobalChat] <Console> %s" % self.args[2:]
```
Comment (alama) [inline proxy/plugins/GlobalChat.py]: you may want to put back the newline here

**Comment by mikotohe2** [comment]
Whoops, my bad.

**Comment by cyberkitsune** [comment]
While I like the idea of this, you can already set up the bot to autoauth to nickserv using the config's autoexec:

**Comment by mikotohe2** [comment]
I suppose, but I feel like actual nickserv features could be more convenient, especially for people that don't know enough about IRC to know about PRIVMSG, etc.

**Comment by acffordyce973** [comment]
I have to agree, this is much more user friendly, especially for those with minimal IRC knowledge. Being able to auth with NickServ easily is usually a feature of most IRC bots too.

**Comment by mikotohe2** [comment]
This should work, tested on Badnik and my network.

**Pull request merged**